recipes/jasper/all/conanfile.py

```diff
@@ -37,25 +45,22 @@ def source(self):
         os.rename(extracted_dir, self._source_subfolder)
 
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["JAS_ENABLE_DOC"] = False
-        cmake.definitions["JAS_ENABLE_PROGRAMS"] = False
-        cmake.definitions["JAS_ENABLE_SHARED"] = self.options.shared
-        cmake.definitions["JAS_LIBJPEG_REQUIRED"] = "REQUIRED"
-        cmake.definitions["JAS_ENABLE_OPENGL"] = False
-        cmake.configure(source_folder=self._source_subfolder)
-        return cmake
-
-    def _patch_cmake(self):
-        cmake_file = os.path.join(self._source_subfolder, "CMakeLists.txt")
-        tools.replace_in_file(cmake_file,
-                              "project(JasPer LANGUAGES C)",
-                              """project(JasPer LANGUAGES C)
-                                 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-                                 conan_basic_setup()""")
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["JAS_ENABLE_DOC"] = False
+        self._cmake.definitions["JAS_ENABLE_PROGRAMS"] = False
+        self._cmake.definitions["JAS_ENABLE_SHARED"] = self.options.shared
+        self._cmake.definitions["JAS_LIBJPEG_REQUIRED"] = "REQUIRED"
+        self._cmake.definitions["JAS_ENABLE_OPENGL"] = False
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
 
     def build(self):
-        self._patch_cmake()
+        os.rename(os.path.join(self._source_subfolder, "CMakeLists.txt"),
```

Comment:
Since you have added a CMake wrapper it shouldn't be necessary to rename the original CMake file

The original file is in `source_subfolder`, the wrapper one directory above

Reply:
Tried that the first time, but I had an error related to include(InSourceBuild)